umapi_client/functional.py

```diff
@@ -308,6 +320,48 @@ def delete_account(self):
         self.append(removeFromDomain={})
         return None
 
+    def maybe_split_groups(self, max_groups):
+        """
+        Check if group lists in add/remove directives should be split and split them if needed
+        :param max_groups: Max group list size
+        :return: True if at least one command was split, False if none were split
+        """
+        new_commands = []
+        # return True if we split at least once
+        maybe_split = False
+        for command in self.commands:
+            step_key, step_args = next(six.iteritems(command))
+            if step_key not in [StepKeys.add.name, StepKeys.remove.name, StepKeys.addRoles.name]:
```

Comment:
This is still a list of constants in code.  Instead change the enum StepKeys to just be GroupStepKeys and just have these three values in the constant.  (Then change this to `step_key in GroupStepKeys`.)  You might also consider having the list be a local constant string, and saving the enum completely.  Especially now that it's local to the UserAction code, it's not so jarring to have it as a constant.

Reply:
What do you mean by local constant string?  Should I just go back to something like `step_key in 'remove', 'add', 'addRole']`?  I thought we were trying to avoid that.

Reply:
Welcome to life with me as your code reviewer :).  Yes, we were definitely trying to avoid that _when the constant string was in a different file from the one where those names mean something_.  Now that all of the code is in `UserAction`, I don't have a problem with the simpler construct, although I think it would be good documentation to have a local variable defined and used in this function, so something like:
```python
keys_that_act_on_group_lists = ['add', 'addRole', 'remove']
... step_key in keys_that_act_on_group_lists
```
Also, be careful about remove: it also takes a string ("all") as its arg, rather than a group list.  So don't apply the length test without first verifying that the `group_list` value is really a list!